styleguide/git/branching-model.md

```diff
@@ -0,0 +1,29 @@
+# Git Branching Model
+
+Joinbox uses a more specific version of the git-flow branching model
+
+![Alt text](git-flow.png "Joinbox GIT Flow")
+
+
+## General
+
+- the master branch can be released at any time, it is always 100% stable
+- the develop branch contains only code that works and is stable but not yet tested by the client
+- features and bug fixes are developed in feature branches until they are stable. the base branch for creating feature branches is the release branch where the issue is in or the master branch for hot fixes.
+- feature branches may only merged into the develop branch if they are tested and work correctly
+- for each iteration (release cycle, milestone) a new release branch is created from develop branch when the development starts.
+- the release branch is published to a testing server for acceptance tests by the client. the client should receive the change log of the release branch
+- fixes for the release branch are merged into the release branch
+- the release branch may be merged into develop or another release branch if it is required
+- hot fixes are only created if they must be applied to the master branch for immediate release. they are based on the master branch
+- after a hot fix or release was released the master branch is merged into the develop branch
+- releases must be planned in advance, there should no more than one release branch at any time
+- after the release branch was merged into master and tagged with the correct version the release branch may be deleted.
+
+## Naming
+Branches must not be updated after a review was requested, before changes are requested in the review.
```

Comment:
I changed this form after opening a PR to after requesting a review form your suggestion.
The Idea behind is to avoid people using time to review code an the the developer updates the code because it was not finished so one needs to review the code again. So basically a way to tell when one need to make a review.